pkg/detector/detector.go

```diff
@@ -767,6 +767,30 @@ func (d *ResourceDetector) BuildResourceBinding(object *unstructured.Unstructure
 		propagationBinding.Spec.ReplicaRequirements = replicaRequirements
 	}
 
+	if features.FeatureGate.Enabled(features.PriorityBasedScheduling) {
+		var bindingSchedulePriority *workv1alpha2.SchedulePriority
+		priorityClassName := policySpec.SchedulePriority.PriorityClassName
+		switch policySpec.SchedulePriority.PriorityClassSource {
+		case policyv1alpha1.KubePriorityClass:
+			kubePriorityClass, err := helper.GetPriorityClassByNameOrDefault(context.TODO(), d.Client, policySpec.SchedulePriority.PriorityClassName)
+			if err != nil {
+				klog.Errorf("Failed to get Kube PriorityClass(%s): %v", priorityClassName, err)
+			}
```

Comment:
```suggestion
			kubePriorityClass, err := helper.GetPriorityClassByName(context.TODO(), d.Client, policySpec.SchedulePriority.PriorityClassName)
			if err != nil {
				klog.Errorf("Failed to get Kube PriorityClass(%s): %v", policySpec.SchedulePriority.PriorityClassName, err)
				return nil, err
			}
```
I think we need to return if the PriorityClass is not found, otherwise it will raise another panic as the `kubePriorityClass` is nil.